packages/icon-button/index.js

```diff
@@ -0,0 +1,129 @@
+import React, {Component} from 'react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import withRipple from '@material/react-ripple';
+import {MDCIconButtonToggleFoundation} from '@material/icon-button';
+
+const {strings} = MDCIconButtonToggleFoundation;
+
+export class IconButtonBase extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      classList: new Set(),
+      childContent: props.children,
+      [strings.ARIA_PRESSED]: props[strings.ARIA_PRESSED],
+      [strings.ARIA_LABEL]: props[strings.ARIA_LABEL],
+    };
+  }
+
+  componentDidMount() {
+    this.initializeFoundation();
+  }
+
+  initializeFoundation = () => {
+    this.foundation_ = new MDCIconButtonToggleFoundation(this.adapter);
+    this.foundation_.init();
+  }
+
+  get classes() {
+    const {classList} = this.state;
+    const {className} = this.props;
+    return classnames('mdc-icon-button', Array.from(classList), className);
+  }
+
+  get adapter() {
+    return {
+      addClass: (className) =>
+        this.setState({classList: this.state.classList.add(className)}),
+      removeClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      // the foundation needs data attributes that will be passed in as props if the user wants this feature
+      getAttr: (attr) => this.props[attr],
+      setAttr: (attr, value) => this.setState({[attr]: value}),
+      setText: (childContent) => this.setState({childContent}),
+    };
+  }
+
+  handleClick_ = (e) => {
+    this.props.onClick(e);
+    this.foundation_.toggle();
```

Comment:
its kinda weird that MDCIconButtonFoundation doesn't have a handleClick method. Instead you are calling `toggle()`. Why doesn't MDCIconButtonFoundation follow our `handleX` pattern?

Reply:
There is a private `toggleFromEvt_` method which is the equivalent `handleX` method. In the web component, it directly calls this `toggle` method. Want to change the private `toggleFromEvt_` method to a public and rename to `handleClick`?

Reply:
This can be updated now

Reply:
I think this comment no longer applies since the MDC Web component has a setter that calls foundation.toggle():
```
  set on(isOn) {
    this.foundation_.toggle(isOn);
  }```